gradle-plugin/src/main/kotlin/dev/zacsweers/metro/gradle/MetroPluginExtension.kt

```diff
@@ -102,6 +102,12 @@ constructor(layout: ProjectLayout, objects: ObjectFactory, providers: ProviderFa
         providers.gradleProperty("metro.version.check").map { it.toBoolean() }.orElse(true)
       )
 
+  /**
+   * Enable/disable suggestion to lift @Inject to class if there is no params. Enabled by default.
+   */
+  public val suggestClassInjectionIfNoParams: Property<Boolean> =
```

Comment:
Let's call it `warnOnInjectAnnotationPlacement` and keep it generic, and update the other names accordingly. Let's also add a test for this to compiler-tests